src/utils/dep-warn.js

```diff
@@ -0,0 +1,33 @@
+// 'new' is optional
```

Comment:
Nit: about the name of the file/function, it feels too abbreviated. We usually type things out.

Also, if we add the function suggested above (to resolve values using the old props if new ones are not provided) then maybe this file does not need to named after the function. It could be `deprecation-utils.js`